internal/controller/pgupgrade/apply.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2021 - 2022 Crunchy Data Solutions, Inc.
```

Comment:
Were you picturing a new package?

Reply:
The `pgupgrade` and `postgrescluster` controllers each have their own reconciler struct upon which the `apply.go` methods (among other logic) are defined, so changing the code so that the controllers share some of that logic does not seem like a trivial code change, AFAICT. Also seems like it would require TODOs in a lot of places... :thinking:

Reply:
Yeah, and the problem with TODOs is they are very easy to lose track of when planning sprints or thinking about priorities, so if we wanted to do this, we should make a ticket in our backlog about it.

But that's an IF, which is why I would consider throwing a TODO in here for now -- or maybe just discussing it here would be enough, so: 

when we first made the upgrade manager, this `apply.go` was largely taken from pgo; and now that we're combining the managers, we're moving this `apply.go` back into the repo where it came from. So to me, this redundancy is more bothersome than it might be from an other/objective POV. Certainly for this ticket, I think this is fine: let's make sure our upgrade works when it's part of this binary. But in the future, will this redundancy be aggravating (i.e., will it produce more work for us in the future if we have to make changes in two places?) or will this redundancy be fine? Saying it out loud, I'm sort of leaning towards doing nothing right now and waiting to see if it's a problem in the future. Any thoughts?

Reply:
@benjaminjb I lean toward the "do nothing for now" position, but I get where you're coming from. It might be clearer how to extract a layer that pgupgrade and postgrescluster rest on from an actual shotgun surgery experience.